datahub_sap_hana/ingestion.py

```diff
@@ -178,3 +203,178 @@ def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]:
                 wu = item.as_workunit()
                 self.report.report_workunit(wu)
                 yield wu
+
+    def get_column_lineage_view_definitions(self, conn) -> Iterable[Tuple[str, str]]:
+        with conn:
+            inspector = inspect(conn)
+            schema: List[Any] = inspector.get_schema_names()  # returns a list
+
+            for schema_name in schema:
+                views = inspector.get_view_names(schema=schema_name)  # returns a list
+
+                for view_name in views:
+                    view_sql = inspector.get_view_definition(view_name, schema_name)
+
+                    if view_sql:
+                        yield (view_name, view_sql)
+
+    def _get_column_lineage_for_view(
+        self, view_name: str, view_sql: str
+    ) -> Tuple[str, List[Node]]:
+        """Extracts the columns and the sql definitions of a downstream view.
+        These are then passed to the lineage function create a lineage graph of the column of an sql query.
+        """
+
+        expression: DerivedTable = parse_one(view_sql)
+        selected_columns: List[str] = expression.named_selects
+
+        view_sql = view_sql.lower()
+        lineages = [
+            lineage(col_name.lower(), view_sql) for col_name in selected_columns
+        ]
+        return view_name, lineages
+
+    def parse_column_name(self, column_name: str):
+        """Removes the table/name and alias from the column name returned by sqlglot, eg: instead of 'hotel.room', it will return 'room'."""
+
+        parsed_name = column_name.split(".")
+        return parsed_name[1]
+
+    def get_column_view_lineage_elements(
+        self,
+    ) -> Iterable[Tuple[Field, List[UpstreamField]]]:
+        """This function returns an iterable of tuples containing information about the lineage of columns in a view.
+        Each tuple contains a downstream field (a column in a view) and a list of upstream fields
+        (columns in other views or tables that are used to calculate/transform the downstream column).
+        """
+
+        conn = self.get_db_connection()
+
+        for view_name, view_sql in self.get_column_lineage_view_definitions(conn):
+            view_name, column_lineages = self._get_column_lineage_for_view(
+                view_name, view_sql
+            )
+            for lineage_node in column_lineages:
+                downstream_fields = Field(name=lineage_node.name)
+
+                upstream_fields_list: List[UpstreamField] = []
+
+                for column in lineage_node.downstream:
+                    upstream_fields = UpstreamField(
+                        name=self.parse_column_name(column.name),
+                        entity=Entity(name=column.source.name, platform="hana"),
```

Comment:
I think the platform value can come from the config maybe? there isn't an issue here _but_ it makes the code easier to copy and paste and reuse in another provider.